### PR TITLE
Add "Disable rule for files?" option

### PR DESCRIPTION
**Pull request by guillaumewuip**
This add an option to disable the selected rule for the files by adding `/* eslint-disable <rule-name> */` as the first line of all matching files

### CODE REVIEW

**Comment by cateland** [comment]
👍

File: src/cli.js [hunk 1]
```diff
@@ -156,6 +163,16 @@ let cli = {
               } else {
                 console.log(chalk.green(`Fixes applied, ${ruleName} is now passing`));
               }
+            } else if (answers.disableFiles) {
+              const ruleReport = nibbler.getRuleResults(report, answers.rule);
+              const files = ruleReport.results
+                .filter(result => result.errorCount > 0)
```
Comment (sgeorgesWexity) [inline src/cli.js]: You can add  _|| result.warningCount > 0_ in filter condition (the warningCount are considered in the other if condition)

**Comment by ljharb** [comment]
Rather than doing it this way and polluting potentially hundreds of files, this should add an overrides section in the root eslint config.

**Comment by guillaumewuip** [comment]
Thanks for your message @ljharb. My use case is actually  to *not* add an override in the estlint config.

When migrating eslint configs or versions, it could be complicated to rewrite correctly all the code based at once. 

The option proposed in this PR allows us to disable some rules in files that we don't want / can't rewrite at that time. Because the eslint config is not changed, no rule override is applied globally to the whole project so new files we be written with the good eslint rules.
Files disabled for specific rules with this new eslint-nibble option can be migrated latter one after the other, if we want to.

**Comment by ljharb** [comment]
@guillaumewuip perhaps you're not familiar with how eslint 5+'s "overrides" feature works. You can designate an explicit list of files, and an explicit override config for them - meaning, you can add a *single section* to your eslint config that turns off just that one rule in just those files, without polluting potentially thousands of files.

**Comment by guillaumewuip** [comment]
Yes, I know that I can override only specific files from the config. Even in this case, it's not what I want.

It's way more tempting for me and my team to rewrite a file to match the new eslint config if we see a `/* eslint-disable ... */`. We can do it on the go, while working on something else in that file while, if files were disabled via the config file, we can't. Because we work on big code bases, we would have to constantly check if the specific file we just opened is also in the huge eslint config override part (something between 500 - 10000 lines according to the migrations I've just done) and for which rule. It's just impossible to do effectively.

I completely understand that it's not what you would do in your situation. That's why I propose to add this functionality as one more eslint-niblle *option*. I'm absolutely not saying that this is the way to go for every eslint migration.

**Comment by ljharb** [comment]
It’s not solely an option if i have to choose “no” every time.

**Comment by guillaumewuip** [comment]
I agree with you that this is more obstrusive that we would like too, even with the default value set to `no` (so with just having to press Enter).

**Comment by ljharb** [comment]
(fwiw there’s also a popular eslint plugin that disables eslint disable comments themselves)

**Comment by guillaumewuip** [comment]
> (fwiw there’s also a popular eslint plugin that disables eslint disable comments themselves)

Have you a link/name to that plugin @ljharb ? I'm interested :)

**Comment by ljharb** [comment]
https://github.com/mysticatea/eslint-plugin-eslint-comments

**Comment by guillaumewuip** [comment]
Thanks!

**Comment by IanVS** [comment]
Thanks for this PR, and for the discussion.  I'm sorry to say I didn't notice it until just now.  I'll give it a review soon, just in the middle of other things at the moment.

**Comment by jtomaszewski** [comment]
Any chance of reviewing / merging this?

In the meantime, if somebody likes the `overrides` approach, I developed a [script](https://github.com/ailohq/eslint-config/blob/master/bin/transform-eslint-errors-into-overrides.js) to convert `eslint ... -f json` output into `overrides: [...]`.

Although, if I were to choose, I think I prefer @guillaumewuip 's approach of having the comment in the file. (IMHO Then there's higher chance that sb will actually notice the comment, and improve the file over time.)

**Comment by IanVS** [comment]
Thanks for the ping.  My current thinking is that this belongs as a separate tool, as it is a bit outside of the purpose of eslint-nibble.  Perhaps it could be added as an option into https://github.com/IanVS/eslint-filtered-fix, would that meet your needs?

**Comment by jtomaszewski** [comment]
Mhm. Doesn't https://github.com/IanVS/eslint-filtered-fix serve the same goal as https://github.com/IanVS/eslint-nibble ?

**Comment by IanVS** [comment]
My natural tendency is to try to push back on new features lest it get too difficult to use.  Perhaps there's a way to do this which makes sense, though.  

Instead of making this a runtime option which would have to be declined/accepted each time, what if this was a cli argument?

**Comment by jtomaszewski** [comment]
Yeah, a cli argument sounds awesome ;)

**Comment by mizdra** [comment]
@guillaumewuip 
FYI: I developed [eslint-interactive](https://github.com/mizdra/eslint-interactive), inspired by eslint-nibble. This tool allows you to disable rules per file or line. It probably achieves what you want with this PR :)

**Comment by IanVS** [comment]
Thanks @mizdra, I'm glad you were able to take the ideas of this project and improve on them.  I'll spend some time checking it out and unless there are features of eslint-nibble which eslint-interactive does not support, I may end up just pointing to yours in the readme and sunsetting eslint-nibble.

In the meantime, I'll close out this PR.

**Comment by mizdra** [comment]
@IanVS Thank you too. eslint-nibble is a great tool that has given me intense inspiration. To the contrary, if eslint-interactive has inspired you, it would be a great honor.

**Pull request closed without merging**